test/src/test_derive.rs

```diff
@@ -160,3 +167,181 @@ fn test_derive_owned_to_variant() -> bool {
 
     ok
 }
+
+#[derive(gdnative::NativeClass)]
+#[inherit(Node)]
+struct FakeNativeClassWithPropertyHooks {
+    #[property(before_set = "Self::before_get_fake_callback")]
+    before_get_bool: bool,
+    #[property(after_set = "Self::after_get_fake_callback")]
+    after_get_bool: bool,
+    #[property(before_set = "Self::before_set_fake_callback")]
+    before_set_bool: bool,
+    #[property(after_set = "Self::after_set_fake_callback")]
+    after_set_bool: bool,
+
+    pub before_get_callback_calls: u32,
+    pub after_get_callback_calls: u32,
+    pub before_set_callback_calls: u32,
+    pub after_set_callback_calls: u32,
+}
+
+#[gdnative_derive::methods]
+impl FakeNativeClassWithPropertyHooks {
+    fn new(_owner: &Node) -> Self {
+        Self {
+            before_get_bool: false,
+            after_get_bool: false,
+            before_set_bool: false,
+            after_set_bool: false,
+            before_get_callback_calls: 0,
+            after_get_callback_calls: 0,
+            before_set_callback_calls: 0,
+            after_set_callback_calls: 0,
+        }
+    }
+
+    fn before_get_fake_callback(_script: &mut Self, _owner: TRef<Node, Shared>) {
+        _script.before_get_callback_calls += 1;
+    }
+
+    fn after_get_fake_callback(_script: &mut Self, _owner: TRef<Node, Shared>) {
+        _script.after_get_callback_calls += 1;
+    }
+
+    fn before_set_fake_callback(_script: &mut Self, _owner: TRef<Node, Shared>) {
+        _script.before_set_callback_calls += 1;
+    }
+
+    fn after_set_fake_callback(_script: &mut Self, _owner: TRef<Node, Shared>) {
+        _script.after_set_callback_calls += 1;
+    }
```

Comment:
Underscores are put in front of identifiers to silence unused variable warnings. They are not mandatory. Also since this is in an impl block, you can use `&mut self` instead of `_script: &mut Self`.